be/src/vec/functions/array/function_array_remove.h

```diff
@@ -0,0 +1,370 @@
+
+        size_t cur = 0;
+        for (size_t row = 0; row < offsets.size(); ++row) {
+            size_t off = offsets[row - 1];
+            size_t len = offsets[row] - off;
+
+            if (len == 0) {
+                // case: array:[], target:1 ==> []
+                // case: array:NULL, target:1 ==> NULL
+                dst_offsets.push_back(cur);
+                continue;
+            }
+
+            size_t count = 0;
+            for (size_t pos = 0; pos < len; ++pos) {
+                if (!nested_null_map && target_null_map[row]) {
+                    // case: array:[1,2], target:Null ==> [1,2]
+                    dst_column.insert_from_not_nullable(src_column, off + pos);
+                    continue;
+                }
+
+                if (nested_null_map) {
+                    if (nested_null_map[off + pos] && !target_null_map[row]) {
+                        // case: array:[Null], target:1 ==> [Null]
+                        dst_column.insert_data(nullptr, 0);
+                        continue;
+                    } else if (!nested_null_map[off + pos] && target_null_map[row]) {
+                        // case: array:[1,2], target:Null ==> [1,2]
+                        dst_column.insert_from_not_nullable(src_column, off + pos);
+                        continue;
+                    } else if (nested_null_map[off + pos] && target_null_map[row]) {
+                        // case: array:[Null], target:Null ==> []
+                        ++count;
+                        continue;
+                    }
+                }
+
+                if (src_data[off + pos] == target_data[row]) {
+                    ++count;
+                } else {
+                    dst_column.insert_from_not_nullable(src_column, off + pos);
+                }
+            }
+
+            cur += len - count;
+            dst_offsets.push_back(cur);
+        }
+
+        return dst;
+    }
+
+    ColumnPtr _execute_string(const ColumnArray::Offsets& offsets, const IColumn& nested_column,
+                              const UInt8* arr_null_map, const IColumn& target,
+                              const UInt8* nested_null_map, UInt8* dst_null_map, UInt8* target_null_map) {
+        // check array nested column type and get data
+        auto& src_column = reinterpret_cast<const ColumnString&>(nested_column);
+
+        // check array nested column type and get data
+        const auto& src_offs = reinterpret_cast<const ColumnString&>(nested_column).get_offsets();
+        const auto& src_chars = reinterpret_cast<const ColumnString&>(nested_column).get_chars();
+
+        // check right column type and get data
+        const auto& target_offs = reinterpret_cast<const ColumnString&>(target).get_offsets();
+        const auto& target_chars = reinterpret_cast<const ColumnString&>(target).get_chars();
+
+        // prepare dst array
+        auto dst = ColumnArray::create(
```

Comment:
need to consider nested column nullable or not.

Reply:
Ok. I have refactored the code.